flink-table/flink-table-common/src/main/java/org/apache/flink/table/sources/PartitionableTableSource.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.sources;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Adds support for partition pruning to a [[TableSource]].
```

Comment:
I do a little change: `A partition is represented as a Map<String, String>` instead of `Partitions are ...`. Because a `Map<String, String>` represents only one partition which is consisted of several field names -> static field values.